### PR TITLE
Merge infrahub-git agent and task runner into task-worker

### LINKED ISSUE #1075: task: Rename the git-agent
### Component

Git Integration

### Task Description

For now "git-agent" is our name for the worker that is subscribing to messages containing tasks that need to be processed asynchronously. As this function works with git repositories it has been given the name git-agent, however, it can also be confusing and misleading as this agent also performs other tasks.

A suggestion would be that we rename it as either only `worker` or `infrahub-worker`. Does anyone have any other suggestions?

### PR DESCRIPTION
**Pull request by dgarros**
Fixes https://github.com/opsmill/infrahub/issues/1075

This PR merge both type of worker `infrahub-git` and `task-runner-internal` into a single worker called `task-worker`
In the process I also renamed `infrahub-server` as `server` to be consistent with the name of the other services

Merging the workers together will help with the migration to prefect and it will keep the overall resource utilization of Infrahub down.

**!! Local override file files might need to be updated if they reference the services `infrahub-server` or `infrahub-git`**

### CODE REVIEW

File: development/docker-compose.yml [hunk 1]
```diff
@@ -140,6 +143,7 @@ services:
       retries: 20
       start_period: 10s
   infrahub-git:
```
Comment (fatih-acar) [inline development/docker-compose.yml:145]: Shouldn't we remove all references to `infrahub-git`?

File: tasks/demo.py [hunk 2]
```diff
@@ -30,6 +31,8 @@
 
 NAMESPACE = Namespace.DEFAULT
 
+SERVICE_WORKER_NAME = "infrahub-git"
```
Comment (fatih-acar) [inline tasks/demo.py:34]: Why is there a different SERVICE_WORKER_NAME?

**Review by fatih-acar — COMMENTED** [review]
I think we should also start updating the `docker-compose.yml` file used for the one liner setup.
And also update the Helm chart. But these can be in another PR.

File: backend/infrahub/workers/infrahub_async.py [hunk 3]
```diff
@@ -61,7 +62,12 @@ async def setup(self, **kwargs: dict[str, Any]) -> None:
         config_file = os.environ.get("INFRAHUB_CONFIG", "infrahub.toml")
         config.load_and_exit(config_file_name=config_file)
 
-        self._logger.debug(f"Using Infrahub API at {config.SETTINGS.main.internal_address}")
+        # Start metric endpoint on port 8000
+        metric_port = int(os.environ.get("INFRAHUB_METRICS_PORT", 8000))
+        self._logger.info(f"Starting metric endpoint on port {metric_port}")
+        start_http_server(metric_port)
+
+        self._logger.info(f"Using Infrahub API at {config.SETTINGS.main.internal_address}")
```
Comment (ogenstad) [inline backend/infrahub/workers/infrahub_async.py:70]: I think we should get rid of this `config.SETTINGS.main.internal_address` setting and instead rely on `INFRAHUB_ADDRESS`, the `internal_address` is defined as an optional string which is incorrect in this context and we already use `INFRAHUB_API_TOKEN` which can't be defined in the infrahub.toml file.

File: development/Dockerfile [hunk 4]
```diff
@@ -76,6 +76,8 @@ RUN mkdir -p /opt/infrahub/git /opt/infrahub/storage /opt/infrahub/source /opt/i
 
 WORKDIR /source
 
+ENV PREFECT_WORKER_QUERY_SECONDS="2"
```
Comment (ogenstad) [inline development/Dockerfile:79]: This feels like something that can easily be missed when we create other containers. Is it something that should live within the code in some way?

**Review by ogenstad — APPROVED** [review]
LGTM. Might be a bit annoying with the rename of the service when switching between `stable` and `develop` with regards to the override files but that should be a short lived problem.

There will be a few things that should be updated in the docs as well and as a follow up step we should consider the `ComponentType` class. and rename the GIT_AGENT enum.

Fixes #1075 :)

File: development/Dockerfile (hunk 4 above)
Comment (dgarros) [reply development/Dockerfile:79]: True, I haven't found a good way to configure prefect other than using Env variables, I'm sure it's possible but I didn't got a chance to nail that down yet.

**Pull request merged**